source/xatlas/xatlas.cpp

```diff
@@ -814,6 +815,12 @@ static float length(const Vector3 &v)
 	return sqrtf(lengthSquared(v));
 }
 
+static float angle(const Vector3 &a, const Vector3 &b)
+{
+	const Vector3 c = cross(a, b);
+	return abs(atan2(length(c), dot(a, b)));
```

Comment:
if you want you can try using a polynomial fit for this. it musnt be precise. look how cute it is when I try the montreal university fit (Sreeraman Rajan et al.)
![image](https://github.com/user-attachments/assets/778514c2-d8d6-4801-a843-68146809e44a)

Reply:
Not so sure, `atan2` should give results in the range [-pi, pi], while `atan` gives results in the range [-pi/2, pi/2].
Apart from that, that smells like premature optimization imo ("no visible slowdown").

Reply:
yes it is. don't touch it, the test showed it's not necessary